File: src/Page/Page.java
package Page;

import java.util.Scanner;
import SystemMessage.*;

public abstract class Page {
	private PageType type;
	private String name;
	private String content; 
	
	
	// constructor
	public Page(String content) {
		setContent(content);
	}
	
	// the abstract method, this will be build later 
	public abstract SystemMessage runPage();
	
	// this function print the content
	public void showContent() {
		System.out.println(getContent());
	}

	public String takeInput() {
		Scanner scanner = new Scanner(System.in);
		String input = scanner.nextLine();
		return input;
	}
	
	// GETTER - SETTER METHODS
	public PageType getType() {
		return type;
	}

	public void setType(PageType type) {
		this.type = type;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}	
}

File: python_development/Page/Page.py
from abc import ABC, abstractmethod

class Page(ABC):
    pageType = None
    name = None
    content = None

    def __init__(self, content):
        self.setContent(content)


    @abstractmethod
    def runPage(self):
        pass


    def showContent(self):
        print(self.content)


    def takeInput(self):
        return input()
    
    
    ### GETTER SETTER METHODS
    def getType(self):
        return self.pageType
    
    def setType(self, pt):
        self.pageType = pt

    def getName(self):
        return self.name
    
    def setName(self, name):
        self.name = name

    def getContent(self):
        return self.content
    
    def setContent(self, content):
        self.content = content